src/cpp/lpnamer/problem_modifier/XpansionProblemsFromAntaresProvider.cpp

```diff
@@ -21,15 +24,30 @@ std::vector<std::shared_ptr<Problem>> XpansionProblemsFromAntaresProvider::provi
 {
     std::vector<std::shared_ptr<Problem>> xpansion_problems;
     xpansion_problems.reserve(antares_hebdo_problems.weekCount());
+    std::vector<std::pair<const Antares::Solver::WeeklyProblemId&,
+                          const Antares::Solver::WeeklyDataFromAntares&>>
+      weekly_problems_vector;
+    weekly_problems_vector.reserve(antares_hebdo_problems.weeklyProblems.size());
     for (const auto& [problem_id, hebdo_data]: antares_hebdo_problems.weeklyProblems)
     {
-        auto&& problem = AntaresProblemToXpansionProblemTranslator::translateToXpansionProblem(
-          antares_hebdo_problems,
-          problem_id.year,
-          problem_id.week,
-          solver_name,
-          solver_log_manager);
-        xpansion_problems.push_back(problem);
+        weekly_problems_vector.emplace_back(problem_id, hebdo_data);
     }
+    std::mutex mutex;
```

Comment:
You probably don't need a `mutex` here if you have a unique index for each iteration
```cpp
std::vector<Problem> out(nbElements);
std::for_each(std::execution::par, in.begin(), in.end(), [&](const auto& pair)
{
    std::size_t index = getIndex(context);
    out[index] = getProblem();
});
```
Although adding a `std::shared_ptr` to a `std::vector` shouldn't take too long

Reply:
We could compute an index from a WeeklyProblemId, but the mutex is clearer and more robust imo